catalog/spatial/registry/registry-rest-endpoint/src/main/java/org/codice/ddf/registry/rest/endpoint/RegistryRestEndpoint.java

```diff
@@ -150,14 +149,114 @@ public Response viewRegInfoHtml(@PathParam("registryId") final String registryId
                     .build();
         }
 
-        Map<String, Object> registryMap = reportMapBuilder.buildRegistryMap(registryPackage);
+        try {
+            html = registryReportBuilder.getHTMLFromRegistryPackage(registryPackage,
+                    templateLoader,
+                    "report");
+        } catch (IOException e) {
+            String message = "Error in compiling applying report template.";
+            LOGGER.error("Error in compiling applying report template");
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR)
+                    .entity(message)
+                    .build();
+        }
+
+        return Response.ok(html)
+                .build();
+    }
+
+    @Path("/{registryId}/report/summary")
+    @Produces(MediaType.TEXT_HTML)
+    @GET
+    public Response viewSummaryInfoHtml(@PathParam("registryId") final String registryId) {
+        String html = "";
+        Metacard metacard;
+
+        if (StringUtils.isBlank(registryId)) {
+            return Response.status(Response.Status.BAD_REQUEST)
+                    .entity("Registry Id cannot be blank")
+                    .build();
+        }
+        try {
+            metacard =
+                    (federationAdminService.getRegistryMetacardsByRegistryIds((Collections.singletonList(
+                            registryId)))
+                            .get(0));
+        } catch (FederationAdminException e) {
+            String message = "Error getting Metacard.";
+            LOGGER.error("{} For registry id: '{}', optional sources: {}", message, registryId);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR)
+                    .entity(message)
+                    .build();
+        }
 
         try {
-            Handlebars handlebars = new Handlebars(templateLoader);
-            Template template = handlebars.compile("report");
-            html = template.apply(registryMap);
+            html = registryReportBuilder.getHTMLFromMetacard(metacard, templateLoader, "summary");
         } catch (IOException e) {
-            LOGGER.error("Error compiling and applying report template.");
+            String message = "Error in compiling applying report template.";
+            LOGGER.error("Error in compiling applying report template");
```

Comment:
Do we want to log message. There was a similar error that read 'compiling and applying' was the 'and' removed on purpose? Do we want to add the registry id to the LOGGER message and say that it is the 'summary' report?